src/main/java/org/gridsuite/loadflow/server/LoadFlowService.java

```diff
@@ -69,33 +70,37 @@ public LoadFlowService() {
 
     }
 
-    private Network getNetwork(UUID networkUuid) {
+    private Network getNetwork(UUID networkUuid, String variantId) {
         try {
-            return networkStoreService.getNetwork(networkUuid, PreloadingStrategy.COLLECTION);
+            Network network = networkStoreService.getNetwork(networkUuid, PreloadingStrategy.COLLECTION);
+            if (variantId != null) {
+                network.getVariantManager().setWorkingVariant(variantId);
```

Comment:
You don't need to set working variant as you use the runner.run(network, variantId, ...) method ?

Reply:
You're right.
Modification done